src/fitnesse/junit/JUnitXMLTestResultRecorder.java

```diff
@@ -0,0 +1,118 @@
+package fitnesse.junit;
+
+import org.apache.commons.text.StringEscapeUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Recorder for test results in the JUnit xml format.
+ * The xml report is written to the file system.
+ */
+public class JUnitXMLTestResultRecorder {
+
+  private final File reportsDir;
+
+  /**
+   * Constructs a JUnitXMLTestResultRecorder.
+   * @param reportsDir directory the test results are to be written to (eg. <code>reports.junitXml.destination</code>)
+   */
+  public JUnitXMLTestResultRecorder(File reportsDir) {
+    this.reportsDir = reportsDir;
+    this.reportsDir.mkdirs();
+  }
+
+  /**
+   * Records result of single page (i.e. test).
+   * @param testName name of test
+   * @param skipped number of skipped tests
+   * @param failures number of test failures
+   * @param errors number of errors of test
+   * @param throwable throwable from test
+   * @param executionTime execution time in seconds
+   */
+  void recordTestResult(String testName, int skipped, int failures, int errors, Throwable throwable, double executionTime) throws
+    IOException {
+    String resultXml = generateResultXml(testName, skipped, failures, errors, throwable, executionTime);
+    writeResult(testName, resultXml);
+  }
+
+  /**
+   * Writes XML result to disk.
+   * @param testName name of test
+   * @param resultXml XML description of test outcome
+   * @throws IOException if unable to write result
+   */
+  private void writeResult(String testName, String resultXml) throws IOException {
+    String finalPath = getXmlFileName(testName);
+    Writer fw = null;
```

Comment:
You can use 'try-with-resources' here and

Reply:
Fixed. Please revise.